### ISSUE TITLE
Disable preserving state

### ISSUE THREAD

**Issue by steve-cahn** [Milad-Akarie/auto_route_library#505]
How can I not preserve the state/page when using `bottomNavigationBuilder`? After I navigate to another tab, then go back to the original one, the state and everything about it is the same, as if I never left it. How can I make the page reload itself when the tab goes to that page?

**Comment by theweiweiway** [comment]
```   
 AutoTabsScaffold(lazyLoad: false,)   
```

**Comment by steve-cahn** [comment]
It's not working... When I go to a nested route, change the tab, then change it back to the original one, I'm still seeing a nested route...

**Comment by bogdanmarinescu** [comment]
this is not a bug,  this should be marked as a feature.

**Comment by theweiweiway** [comment]
@bogdanmarinescu It's a bug because the feature is implemented but not working as intended

**Comment by bogdanmarinescu** [comment]
> @bogdanmarinescu It's a bug because the feature is implemented but not working as intended

What is the intended behavior? If I move from one BottomNavigationBarItem to another the previous tab should get disposed?

**Comment by theweiweiway** [comment]
I see - sorry i misunderstood the question, 

The `lazyload` is only for the initial load. Not switching back and forth between tabs and disposing the tab when exited. Yes, it is a feature

**Comment by mikepyts** [comment]
One of the cases needed for me is that I am loading useful information each time in initState() of the selected tab route page, but with the preserved state, I need to move this into another place, bottom tab navigation bar etc.

**Comment by Milad-Akarie** [comment]
@mikepyts  seems to me you just want to notify your tab overtime it's re-selected is that correct?

**Comment by Milad-Akarie** [comment]
The whole idea of tabs router is to maintain state in parallel, if you don't want to preserve your state for any of your tabs just useqn AutoRouter() instead and call router.replace in onTabSelected

**Comment by mikepyts** [comment]
@Milad-Akarie sorry for misunderstanding, just gave an example in which situations do we use that, of course replacing AutoRoute in tab selection is a possible way, this is all about improvements instead of workarounds and new features. I have found this issue because had the same situation and documentation doesn't cover such situations

**Comment by Milad-Akarie** [comment]
@mikepyts I understand, If i'm to do something on my site it's gonna be a flag that disables state preserving for all tabs which I don't think is a wanted behavior but it's still doable. on the other hand you users can simply disable state preserving for certain routes by assigning a simple key 

```dart
class MyStatefulTabPage extends StatefulWidget {
  MyStatefulTabPage() : super(key: UniqueKey());
....
}
```
MyStatefulTabPage will be rebuilt everytime it's selected.
if you guys wanna go with an overall flag just let me know.

**Comment by steve-cahn** [comment]
@Milad-Akarie  Yes, I think it's nice to have that option of having an overall flag

**Comment by olegys** [comment]
> @mikepyts I understand, If i'm to do something on my side it's gonna be a flag that disables state preserving for all tabs which I don't think is a wanted behavior but it's still doable. on the other hand you users can simply disable state preserving for certain routes by assigning a simple key

This solution dont work...

**Comment by Maadhav** [comment]
#847 This seems to be a fix for the persistence problem

**Comment by breucode** [comment]
> @mikepyts I understand, If i'm to do something on my side it's gonna be a flag that disables state preserving for all tabs which I don't think is a wanted behavior but it's still doable. on the other hand you users can simply disable state preserving for certain routes by assigning a simple key
> 
> ```dart
> class MyStatefulTabPage extends StatefulWidget {
>   MyStatefulTabPage() : super(key: UniqueKey());
> ....
> }
> ```
> 
> MyStatefulTabPage will be rebuilt everytime it's selected. if you guys wanna go with an overall flag just let me know.

Unfortunately, this does not work for me. I have a Stateful Widget, with a unique key, but the state is never updated.
